### PR TITLE
UI: Fix source tree item color height

### PR DESCRIPTION
**Pull request by cg2121**
### Description
Fixes widget inside of source tree item being incorrect height.

Before:
Screenshot from 2022-08-07 01-47-41

After:
Screenshot from 2022-08-07 02-11-05

### Motivation and Context
Fix theme bugs

### How Has This Been Tested?
Set scene item colors and made sure everything looked good.

### Types of changes
- Bug fix (non-breaking change which fixes an issue)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by WizardCM** [comment]
As we're partially breaking existing themes as part of 28 anyway, what if we just restructure this widget so that hacks like this aren't necessary any more? We've had to fight this particular element in various ways for ages.

**Comment by cg2121** [comment]
We can't restructure the widget, as it is a QT problem that it works this way, afaik.

**Comment by cg2121** [comment]
According to a bunch of forum posts, the only way to fix this without changing the qss would to create a delegate and override the paint event in the source tree, which I don't feel like doing. Just some dumb QT things.

**Comment by WizardCM** [comment]
Ah, that's a bummer. Thanks for looking into it.

**Comment by SuslikV** [comment]
The margins should be set in the .qss, not in the code.
Example, 3px from all sides:
```
SourceTree {
	margin: 3px;
}
```
At some point, you should decide either you using .qss or customizing the whole UI manually, in the code.

The padding works OK. Probably, you misunderstanding the box model of the widget. It is not known what the theme should look like (the author should say if the selection border wasn't intentional).

Also, please fix the "group"ed sign width (obviously, it is cut in Yami theme):
At least, like this:
```
SourceTreeSubItemCheckBox {
...
	min-width: 12px;
}
```

**Comment by SuslikV** [comment]
Also, the string:
https://github.com/obsproject/obs-studio/blob/61e0d69779882fdbca2f38856237c7cd0bc4ca86/UI/data/themes/Yami.qss#L184
do nothing. It is seems to be leftover from the attempt to round the background color from the _Set Color_, isn't?

If so, then the:
```
*[bgColor="1"]
{
	border-radius: 4px;
}
```
may help (should be set for all 8 colors)

If not - the string can be removed (only Acri theme uses border here, and the style isn't preserved on theme change, for me at least).

**Review by Warchamp7 — APPROVED** [review]
Seems like the lesser of all evils for fixing this.

Pending code review I am fine with this being fixed.

**Comment by cg2121** [comment]
Fixed merge conflicts and restored the padding in the scene tree.

**Comment by GeorgesStavracas** [comment]
Can source rows not lose their rounded corners? Can the scene list not lose its current padding?

**Comment by GeorgesStavracas** [comment]
Apparently the hardcoded heights do not scale on DPI changes.

**Review by Warchamp7 — CHANGES_REQUESTED** [review]
We'll need to find a different approach that doesn't involve hardcoding height, so as not to break due to DPI scaling

**Comment by cg2121** [comment]
@GeorgesStavracas could you post a screenshot of what they look like under high dpi?

**Comment by cg2121** [comment]
I finally figured out a way of doing this without hardcoding the heights.

@GeorgesStavracas could you confirm it works with different dpis?

**Comment by cg2121** [comment]
Fixed merge commits.

**Comment by RytoEX** [comment]
Here's how this PR currently looks on my laptop screen, which uses a 125% custom scaling factor.

image

**Comment by GeorgesStavracas** [comment]
Thanks @RytoEX, this looks good to me now - I wonder if the QSS changes here affect the menu style though, @cg2121 could you confirm menus still have the same style as of lists?

**Review by Warchamp7 — APPROVED** [review]

**Comment by cg2121** [comment]
@GeorgesStavracas the menu is styled correctly.

Screenshot from 2022-08-24 21-38-18

**Comment by RytoEX** [comment]
What was the purpose of this change after an approval was given? It looks like some stylistic preference rather than a functional change, so it _shouldn't affect anything, unless there's some importance to the order of identifiers here. At minimum, please point out that you pushed a change and what it was for so we don't miss the fact that a change was pushed after approval, so that we can ascertain if we need to retest or re-review.

https://github.com/obsproject/obs-studio/compare/2ae773a39ecf328c1db79f873450e81aeec7a430..76e8c3da14468c480afcfd9326b9bf85642183ef

**Comment by cg2121** [comment]
It was because of a bad rebase, as I didn't realize a line was removed.

**Comment by RytoEX** [comment]
> It was because of a bad rebase, as I didn't realize a line was removed.

Ah, I see. Some lines were reordering, some were adding in the accidentally removed lines.

File: UI/source-tree.hpp
Comment (SuslikV) [inline UI/source-tree.hpp:29]: Changing the UI code after the release candidate issued is bad practice.

**Pull request merged**
